python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -144,6 +150,11 @@ def asset(
         output_required (bool): Whether the decorated function will always materialize an asset.
             Defaults to True. If False, the function can return None, which will not be materialized to
             storage and will halt execution of downstream assets.
+        version (Union[bool, str]): (Experimental) If `false` is passed, the asset is unversioned.
```

Comment:
This explanation refers to concepts that might not have an obvious meaning for readers.

Thoughts trying to phrase it in terms of the specific impact that setting it will have?

E.g. something like:

> If an asset has an op version that's different than the op version used for its latest materialization, then it and all downstream assets will be considered stale. Conversely, if two materializations for the same asset used the same op version and had the same upstream inputs, then the second materialization won't mark any new downstream assets stale.

Also accepting boolean values seems unintuitive to me. Thoughts on accepting and defaulting to `None` instead? Apologies if we've already discussed this and I'm re-treading old ground.

Reply:
This PR is prior to logical versions on materializations, which are upstack, so further explanation on this arg should be in one of those PRs.